docs/wiki/3_dev_guide/Distribution-Mechanism.md

```diff
@@ -8,32 +8,54 @@ For the tutorial of distributed training, please see [here](../2_user_guide/Dist
-
 ```
           RAF IR
              │
              │ Executed by
              ▼
-    RAF VM / Interpreter
-             │
-             │ Dispatch to
-             ▼
-┌───RAF Op Implementation
-│     (in NCCL dialect)
-│            │
-│            │ Request
-│            ▼             Init
-│    RAF NCCLCommunicator◄──────RAF MPICommunicator
-│            │                            │
-│ Call       │ Provide Handle             │ Call
-│            ▼                            ▼
-└────►NCCL Collective Op          MPI Library APIs
+    RAF VM / Interpreter            Client Script
+             │                            │
+             │ Dispatch to                │ Set
+             ▼                            ▼
+┌───RAF Op Implementation       RAF VoidCommunicator
+│     (in NCCL dialect)                   │
+│            │                            │ Be
+│            │ Request                    │
+│            ▼             Init           ▼               Be
+│    RAF NCCLCommunicator◄──────RAF Global Communicator◄──────RAF MPICommunicator (default)
+│            │                                                       │
+│ Call       │ Provide Handle                                        │ Call
+│            ▼                                                       ▼
+└────►NCCL Collective Op                                      MPI Library APIs
 
 
                  Architecture Diagram
 ```
 
+Distributed RAF mainly adopts the Single Program Multiple Data (SPMD) programming model driven by MPI-like libraries such as NVIDIA Collective Communications Library (NCCL), which provides a set of collective communication operators and bridges multiple discrete GPUs on numerous servers connected via conventional Ethernet or high-performance Infiniband. By default, RAF uses MPI to launch the Python script. In such a situation, RAF utilizes both MPI and NCCL in a mixed way, where MPI is used to launch the distributed RAF (create many MPI processes) on user-specified nodes and exchange data for initializing NCCL, while NCCL is responsible for transferring and computing tensors among multiple GPUs efficiently.
+
+RAF also offers the ability of using other launchers instead of MPI. RAF provides a set of API to set/get the global communicator. By default, it will be a `MPICommunicator`. Users can switch to use `VoidCommunicator`, so that users can change those distributed infomation freely and use other launchers (e.g. DeepSpeed, torchrun, user-owned multi-process launcher, etc.). When using `VoidCommunicator`, the data exchange for initializing NCCL will be done by using shared file(s), so environment variable `RAF_FILE_STORE_PATH` should be set to a directory.
+
+*P.S.: `VoidCommunicator` now supports single machine only.*
```

Comment:
According to this description, am I understanding correctly?
1. When we use VoidCommunitator, we could set size/rank/local_rank by ourselves to use less number of GPUs.
2. It supports single machine because it uses local disk files to exchange data.
3. If my understanding in (2) is correct, maybe we could have a better name for this communicator, like LocalCommunicator, FileCommunicator, etc.

Reply:
1. yes
2. no, the file can be on NFS. it's due to `host_ids` not sync for `VoidCommunicator`. There are 2 solutions: 1) expose `GetHostId` to python, and make `host_ids` set by user manually; 2) add a method to `VoidCommunicator` to also use the file store to sync `host_ids`, it should be call by user or `InitSubCommunicator`
3. I think "void" means it's not related to any communication library (mpi/nccl), and we could use other way to sync data (e.g. a remote kvstore or a tcpstore like pytorch) not just file.

Reply:
Got it. Thanks for the clarification.